### PR TITLE
feature: ✨ intercom-set

### PR DESCRIPTION
**Pull request by NewCodeOnDaBlock**
### Feature

* \[x] Added entry to ./changelog
 * \[x] Link to issue:
* \[x] Add PR label

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -61,9 +61,14 @@ new Vue({
   apolloProvider,
   vuetify,
   beforeCreate() {
-    this.$intercom.boot({
-      user_id: uuidv4()
-    });
+    const userId = this.$route.query.intercomid
```
Comment (gamalielhere) [inline src/main.js:64]: I would check whether intercomid exists here as well.

File: src/main.js [hunk 2]
```diff
@@ -61,9 +61,14 @@ new Vue({
...
+      ? this.$route.query.intercomid
+      : uuidv4();
+    if (userId) {
```
Comment (gamalielhere) [inline src/main.js:67]: won't need to check this since if intercomid is not there, it will be filled with the uuidv4() function

**Review by gamalielhere — CHANGES_REQUESTED** [review]

File: src/main.js [hunk 3]
```diff
@@ -61,9 +61,13 @@ new Vue({
   apolloProvider,
   vuetify,
   beforeCreate() {
-    this.$intercom.boot({
-      user_id: uuidv4()
-    });
+    const userId = this.$route.query.intercomid
+      ? this.$route.query.intercomid
+      : uuidv4();
+    !this.$route.query.intercomid
```
Comment (gamalielhere) [inline src/main.js:67]: This is unnecessary as you already checked it the line before. Just call boot with userId.

**Review by gamalielhere — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 3 above)
Comment (NewCodeOnDaBlock) [reply src/main.js]: fixed

**Review by gamalielhere — APPROVED** [review]

**Review by andrewdahn — APPROVED** [review]

**Review by hsurf22 — APPROVED** [review]

**Pull request merged**